crypto/src/lib.rs

```diff
@@ -403,6 +328,25 @@ impl KeyPair {
     }
 }
 
+#[cfg(feature = "std")]
+impl<'de> Deserialize<'de> for KeyPair {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        use serde::de::Error as _;
+
+        #[derive(Deserialize)]
+        struct KeyPair {
+            public_key: PublicKey,
+            private_key: PrivateKey,
+        }
+
+        let key_pair = KeyPair::deserialize(deserializer)?;
+        Self::new(key_pair.public_key, key_pair.private_key).map_err(D::Error::custom)
+    }
+}
```

Comment:
I don't get it. Original `KeyPair` has the exact same fields. What does this custom impl does that can't be done with `Deserialize` derive?

Reply:
To future readers. Roman answered in DM that it's done to use the `new()` method to check keys consistency.

In that case I suggest to note this in the code and/or comments. I.e. we can remove internal `KeyPair` to `KeyPairUnchecked`.

Reply:
I thought it was obvious from the signature. IMO KeyPairCandidate is a bit better, and a comment just above ::new to emphasise the consistency check would help.